### ISSUE TITLE
pgo 4.2.1: Error is not returned in case a policy contains invalid sql statements

### ISSUE THREAD

**Issue by jose-joye** [CrunchyData/postgres-operator#1266]
Up to version 4.1.x, in case a bad sql script was used inside a policy an error was returned when the policy was applied.

With version 4.2.1, the policy is blindly executed and no error is reported back to the user.

example:
```
bad_script.sql:
   Arggggg, I'm a bad script....;

# creation of policy
pgo -ncfae-xxxx create policy badPolicy --in-file=./bad_script.sql

# version 4.2.1
$ pgo -n cfae-xxxx apply badpolicy --selector=name=jose
Applied policy on jose

#version 4.1.x
$ pgo -ncfae-bit-cbcd apply  badpolicy --selector=name=jose
Error: SQL error: ERROR:  syntax error at or near "rggggg"
LINE 1: rggggg, I'm a bad script....;
        ^
```

Is this an expected behavior with version 4.2.1?
If yes, this makes the policy feature not so useful. We should be able to get a feedback in case the script was not successfully run.

Thanks,
José

**Comment by ajoyej** [comment]
Hello,
Any feedback on this?

We actually use this feature to create dBs and execute small configuration scripts.
For us, this is an important feature. In case the support is dropped, we need to look for another way to achieve our this

Thanks,
José

**Comment by ajoyej** [comment]
I tried to locate the source of the problem....
It seems that changing the  \kubeapi\exec.go the following way will fix the issue (so far, I did not find any side effect):
````
func ExecToPodThroughAPI(config *rest.Config, clientset *kubernetes.Clientset, command []string, containerName, podName, namespace string, stdin io.Reader) (string, string, error) {

        // Code removed for clarity
        .......
       
	if err != nil {
		log.Error(err)
		return stdout.String(), stderr.String(), err
	}

        // Here we check if something is written to stderr. If yes, we had a problem with the script
	if stderr.String() != "" {
		err = fmt.Errorf("error in sql cmd: %s", stderr.String())
		log.Error(err)
		return stdout.String(), stderr.String(), err
	}

	return stdout.String(), stderr.String(), nil

````

José

**Comment by jkatz** [comment]
Hi @jose-joye,

Thanks for working through this one. I made a fix and applied it back down to 4.2. Thanks again for reporting!
